File: content/problems/0349-intersection-of-two-arrays/solutions/set-intersection/Solution.java
class Solution {
    public int[] intersection(int[] nums1,int[] nums2){
        Set<Integer> first=new HashSet<>();
        for(int value:nums1)first.add(value);
        Set<Integer> common=new TreeSet<>();
        for(int value:nums2)if(first.contains(value))common.add(value);
        return common.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: content/problems/0349-intersection-of-two-arrays/solutions/set-intersection/solution.cpp
class Solution {
    public: vector<int> intersection(vector<int>& nums1,vector<int>& nums2){
        unordered_set<int> first(nums1.begin(),nums1.end());
        set<int> common;
        for(int value:nums2)if(first.count(value))common.insert(value);
        return vector<int>(common.begin(),common.end());
    }
};
